iot-hub/Samples/service/CleanUpDevicesSample/CleanUpDevicesSample.cs

```diff
@@ -1,113 +1,149 @@
+                    {
+                        string deviceId = twinResult.DeviceId;
+                        foreach (string prefix in _deleteDeviceWithPrefix)
+                        {
+                            if (deviceId.StartsWith(prefix))
+                            {
+                                _devicesToDelete.Add(new Device(deviceId));
+                            }
+                        }
+                    }
+                }
+
+                var _bulkDeleteList = new List<Device>(DeleteBatchSize);
+                while (_devicesToDelete.Count > 0)
+                {
+                    int i;
+                    for (i = 0; (i < DeleteBatchSize) && (i < _devicesToDelete.Count); i++)
+                    {
+                        _bulkDeleteList.Add(_devicesToDelete[i]);
+                        Console.WriteLine($"\tRemove: {_devicesToDelete[i].Id}");
+                    }
+
+                    _devicesToDelete.RemoveRange(0, i);
+
+                    BulkRegistryOperationResult ret = await _rm.RemoveDevices2Async(_bulkDeleteList, true, CancellationToken.None).ConfigureAwait(false);
+                    devicesDeleted += _bulkDeleteList.Count - ret.Errors.Length;
+                    Console.WriteLine($"BATCH DELETE: {devicesDeleted}/{_bulkDeleteList.Count}");
+                    if (!ret.IsSuccessful)
+                    {
+                        foreach (DeviceRegistryOperationError error in ret.Errors)
+                        {
+                            Console.WriteLine($"\tERROR: {error.DeviceId} - {error.ErrorCode}: {error.ErrorStatus}");
+                        }
+                    }
+
+                    _bulkDeleteList.Clear();
+                }
+
+                Console.WriteLine($"-- Total no of devices deleted: {devicesDeleted}");
+            
+                var configurations = await _rm.GetConfigurationsAsync(100, new CancellationToken()).ConfigureAwait(false);
+                {
+                    foreach (var configuration in configurations)
+                    {
+                        string configurationId = configuration.Id;
+                        foreach (var prefix in _deleteConfigurationWithPrefix)
+                        {
+                            if (configurationId.StartsWith(prefix))
+                            {
+                            _configurationsToDelete.Add(new Configuration(configurationId));
+                            }
+                        }
+                    }
+                }
+                
+                var removeConfigTasks = new List<Task>();
+                _configurationsToDelete.ForEach(configuration =>
+                {
+                    Console.WriteLine($"Remove: {configuration.Id}");
+                    removeConfigTasks.Add(_rm.RemoveConfigurationAsync(configuration.Id));
+                });
+
+                Task.WaitAll(removeConfigTasks.ToArray());
```

Comment:
this Task needs to be awaited:
`await Task.WaitAll(removeConfigTasks.ToArray()).ConfigureAwait(false);`

Reply:
My bad, I read it as Task.WhenAll(...), which will need to be awaited on